imap_processing/hi/l1a/science_direct_event.py

```diff
@@ -166,52 +161,80 @@ def break_into_bits_size(binary_data: str):
     return [binary_data[i : i + 48] for i in range(0, len(binary_data), 48)]
 
 
-def create_dataset(de_data_list: list):
+def create_dataset(de_data_list: list, packet_met_time: list):
     """Create xarray dataset.
 
     Parameters
     ----------
     de_data_list : list
         Parsed direct event data list
+    packet_met_time : list
+        List of packet MET time
 
     Returns
     -------
     dataset: xarray.Dataset
         xarray dataset
     """
     # These are the variables that we will store in the dataset
-    met_subseconds = []
-    met_seconds = []
+    de_met_time = []
     esa_step = []
     trigger_id = []
     tof_1 = []
     tof_2 = []
     tof_3 = []
     de_tag = []
-    for event in de_data_list:
-        # TODO: how to check for fist event?
-        # metaevent.
-        # How to handle if first event is not
-        # metaevent? This means that current
-        # data file started with direct event.
-        # metaevent is a way to store information
-        # about bigger portion of time information. Eg.
-        # metaevent stores information about, let's say
-        # "20240319T09:30:01". Then direct event time
-        # tag stores information of time ticks since
-        # that time. Then we use those two to combine and
-        # get exact time information of each event.
+    ccsds_met = []
+
+    # How to handle if first event is not metaevent? This
```

Comment:
should each direct event always have a metaevent associated with it? Or is it only an issue if the first event in a packet is missing an metaevent?

Reply:
It's only an issue with first event in the first packet of the data file.